src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -109,20 +119,18 @@ public bool SnippetShortcutExists_NonBlocking(string shortcut)
         public virtual bool ShouldFormatSnippet(SnippetInfo snippetInfo)
             => false;
 
-        private void PopulateSnippetCaches()
+        private async Task PopulateSnippetCachesAsync()
         {
             Debug.Assert(_expansionManager != null);
-
-            var token = _waiter.BeginAsyncOperation(GetType().Name + ".Start");
+            AssertIsForeground();
 
             // In Dev14 Update2+ the platform always provides an IExpansion Manager
             var asyncExpansionManager = (IExpansionManager)_expansionManager;
             // Call the asynchronous IExpansionManager API from a background thread
-            Task.Factory.StartNew(() => PopulateSnippetCacheAsync(asyncExpansionManager),
+            await Task.Factory.StartNew(() => PopulateSnippetCacheAsync(asyncExpansionManager),
                             CancellationToken.None,
                             TaskCreationOptions.None,
-                            TaskScheduler.Default).CompletesAsyncOperation(token);
-
+                            TaskScheduler.Default).ConfigureAwait(false);
```

Comment:
```suggestion
            await TaskScheduler.Default;
            await PopulateSnippetCacheAsync(asyncExpansionManager);
```

Reply:
Ah, I tried awaiting the PopulateSnippetCacheAsync regularly, but it didn't go to the background thread - is that what await TaskScheduler.Default is for?

Reply:
Yes, it switches back to a background thread, and then awaits there. 😄